### PR TITLE
SC-7773 - stabilises rabbitMQ connection

### PR DESCRIPTION
**Pull request by virgilchiriac**
# Description

The load on the system, coupled with expiration times, makes rabbitMQ drop connections, which leads to K8 pod restarts every few seconds.

In this PR we reduce the number of times a file is attempted to be downloaded, as well as the sleep time. This decrease the blocking connection from ~31 seconds to ~4 seconds.
At the same time, the heartbeat for rabbitMQ is set to be 120 seconds (default is 60)
The client also specifies heartbeat and timeout which can be configured via secrets/config.yml, as [recommended by pika](https://pika.readthedocs.io/en/stable/examples/heartbeat_and_blocked_timeouts.html)

## Links to Tickets or other pull requests
https://ticketsystem.hpi-schul-cloud.org/browse/SC-7773

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Review by Parnswir — APPROVED** [review]

**Review by mamutmk5 — CHANGES_REQUESTED** [review]
Where is the place where the exceptions who where thrown catched ?

File: antivirus_service/handler.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def scan(self, download_uri, access_token):
                 last_exception_message = str(e)
                 logging.info('file could not downloaded: for {0} time'.format(i))
                 if i < count:
-                    time.sleep(2**i)
+                    time.sleep(2)
         else:
             raise Exception('File could not downloaded: {0} - after {1} seconds'.format(last_exception_message, 2**(i+1) - 1))
```
[the remark this answers is not in the document]
Comment (virgilchiriac) [reply antivirus_service/handler.py]: when this exception is raised, is not gonna scan the file anymore, but the rabbitMQ connection will not terminate, and new requests can be processed

**Review by mamutmk5 — CHANGES_REQUESTED** [review]
Okay if pika lost the rabbitmy connection it don't reconnect.
See https://github.com/pika/pika#connection-recovery how to do it

**Comment by virgilchiriac** [comment]
branch renamed, new PR https://github.com/hpi-schul-cloud/antivirus_check_service/pull/5

**Pull request closed without merging**